internal/dataplane/kongstate/kongstate_test.go

```diff
@@ -45,6 +47,8 @@ var serviceTypeMeta = metav1.TypeMeta{
 
 func TestKongState_SanitizedCopy(t *testing.T) {
 	testedFields := sets.New[string]()
+	// this needs a static random seed because some auths generate random values
+	uuid.SetRand(rand.New(rand.NewSource(1))) //nolint:gosec
```

Comment:
```suggestion
```

I think it would be cleaner to reset the seed for every test case run (as done below), but hardcode the expected output, like:

```diff
 func TestKongState_SanitizedCopy(t *testing.T) {
        testedFields := sets.New[string]()
-       // this needs a static random seed because some auths generate random values
-       uuid.SetRand(rand.New(rand.NewSource(1))) //nolint:gosec
        for _, tt := range []struct {
                name string
                in   KongState
@@ -90,7 +88,7 @@ func TestKongState_SanitizedCopy(t *testing.T) {
                                        Plugin:              kong.Plugin{ID: kong.String("1"), Config: kong.Configuration{"secret": *redactedString}},
                                }},
                                Consumers: []Consumer{{
-                                       KeyAuths: []*KeyAuth{{kong.KeyAuth{ID: kong.String("1"), Key: randRedactedString()}}},
+                                       KeyAuths: []*KeyAuth{{kong.KeyAuth{ID: kong.String("1"), Key: kong.String("{vault://52fdfc07-2182-454f-963f-5f0f9a621d72}")}}},
                                }},
                                Licenses: []License{{kong.License{ID: kong.String("1"), Payload: redactedString}}},
                                ConsumerGroups: []ConsumerGroup{{
```

It wasn't clear to me at first glance that we reset the seed before defining test cases to make `randRedactedString()` calls' results deterministic for test case definitions.

Reply:
Can we do so/do you have code that does it successfully?

I thought we were stuck with start of test and start of case execution resets because we can't reset between individual cases in the case definition, so we need to do all of them in the same sequence.

We can maybe restructure the test somewhat so that we set up cases and then inject values into the cases as we run them, but that felt maybe messier than the currently proposed approach.

Reply:
I think that @pmalek [suggestion](https://github.com/Kong/kubernetes-ingress-controller/pull/5964#discussion_r1593709462) would be the cleanest way.